src/components/animatedImage/index.tsx

```diff
@@ -0,0 +1,74 @@
+// TODO: consider unify this component functionality with our Image component
+import React, {useState, useCallback} from 'react';
+import {StyleSheet, StyleProp, ViewStyle, NativeSyntheticEvent, ImageLoadEventData} from 'react-native';
+import Animated, {FadeIn} from 'react-native-reanimated';
+import View from '../../components/view';
+import Image, {ImageProps} from '../../components/image';
+
+const UIAnimatedImage = Animated.createAnimatedComponent<ImageProps>(Image);
```

Comment:
Is calling this method expensive?

Reply:
Anyone importing a type from this file, and not using `import type` will call this method

Reply:
I do not know, but this is a semi-common usage (from what I've seen in another lib this was a HOC).
I think this is only called once for the whole app, am I wrong? So unless you mean that a public user might accidentally do that, it does not matter.